app/jobs/moab_replication_audit_job.rb

```diff
@@ -1,43 +1,34 @@
-# Confirms that a CompleteMoab is fully/properly replicated to all target zip endpoints.
-# Usage info:
-# MoabReplicationAuditJob.perform_later(cm)
+# Confirms a CompleteMoab has all versions/parts replicated for each of its target endpoints.
+# @note Enqueues a check per endpoint
+# @example usage
+#   MoabReplicationAuditJob.perform_later(cm)
 class MoabReplicationAuditJob < ApplicationJob
   queue_as :moab_replication_audit
-  delegate :check_child_zip_part_attributes, to: Audit::CatalogToArchive
-  delegate :check_aws_replicated_zipped_moab_version, to: PreservationCatalog::S3::Audit
-  delegate :logger, to: Audit::CatalogToArchive
 
   # @param [CompleteMoab] verify that the zip exists on the endpoint
   def perform(complete_moab)
-    druid = complete_moab.preserved_object.druid
-
-    results = AuditResults.new(druid, nil, complete_moab.moab_storage_root, "MoabReplicationAuditJob")
-
-    backfill_missing_zmvs(complete_moab, results) if Settings.replication.audit_should_backfill
-
-    complete_moab.zipped_moab_versions.each do |zmv|
-      next unless check_child_zip_part_attributes(zmv, results)
-      check_aws_replicated_zipped_moab_version(zmv, results)
-    end
-
+    backfill_missing_zmvs(complete_moab)
+    ZipEndpoint
+      .includes(:zipped_moab_versions)
+      .where(zipped_moab_versions: { complete_moab: complete_moab }).each do |endpoint|
+        PartReplicationAuditJob.perform_later(complete_moab, endpoint)
+      end
     complete_moab.update(last_archive_audit: Time.current)
-
-    results.report_results(logger)
   end
 
   private
 
-  def backfill_missing_zmvs(complete_moab, results)
-    backfilled_zmvs = complete_moab.create_zipped_moab_versions!
-    return if backfilled_zmvs.empty?
-
-    results.add_result(
-      AuditResults::ZMV_BACKFILL,
-      version_endpoint_pairs: format_backfilled_zmvs(backfilled_zmvs)
+  def backfill_missing_zmvs(complete_moab)
+    return unless Settings.replication.audit_should_backfill
```

Comment:
Curious about why the choice to check for backfilling here instead of at the call of the method, as before?

Reply:
i think this is probably a bit more OO than my prior approach.  just call the method, and let it determine whether there's anything to be done.

or at least, that was my read on it.

Reply:
It leaves less responsibility in `#perform` to test.  Now the spec for that method just has to test "does `backfill_missing_zmvs` get called"?  The behavior of the other two (logical) lines is more important to the definition of the job and the `#perform` spec can focus on that.